src/classes/CRLP_Rollup_SoftCredit_LDV_BATCH.cls

```diff
@@ -0,0 +1,325 @@
+
+    /** @description The last Parent Id processed by the batch job */
+    private Id lastContactIdProcessed;
+    private Id lastAccountIdProcessed;
+
+    /*******************************************************************************************************
+    * @description Constructs the class, sets the properties
+    */
+    public CRLP_Rollup_SoftCredit_LDV_BATCH() {
+        super();
+        this.isLdvBatchJob = true;
+    }
+
+    /*******************************************************************************************************
+    * @description Constructs the class, sets the properties
+    */
+    public CRLP_Rollup_SoftCredit_LDV_BATCH(List<Id> recordIds) {
+        super(recordIds);
+        this.isLdvBatchJob = true;
+    }
+
+    /*******************************************************************************************************
+    * @description Build a query of OpportunityContactRole records sorted by Contact.AccountId, ContactId, OppId
+    * @param bc BatchableContext passed by the system call
+    * @return database.Querylocator
+    */
+    public database.Querylocator start(Database.BatchableContext bc) {
+
+        String query;
+        String emptyQuery = 'SELECT Id FROM OpportunityContactRole LIMIT 0';
+
+        // If a set of specific Contact or Account Id's was passed, then ONLY roll up to that Object type
+        if (this.recordIds != null && !this.recordIds.isEmpty()) {
+            if (this.recordIds[0].getSobjectType() == Contact.SObjectType) {
+                this.hasAccountSoftCreditRollups = false;
+            } else {
+                this.hasContactSoftCreditRollups= false;
+            }
+        }
+
+        if (this.hasContactSoftCreditRollups == false && this.hasAccountSoftCreditRollups == false) {
+            // If there are no active rollups for the object type, then force the query to return zero records.
+            // This is needed because can't return null as a valid iterator.
+            return Database.getQueryLocator(emptyQuery);
+        }
+
+        try {
+
+            // Build a query of OpportunityContactRole records as the primary object for this batch job
+            List<SObjectType> childObjects = new List<SObjectType>();
+            SObjectType detailObjectType = OpportunityContactRole.SObjectType;
+
+            query = CRLP_Query_SEL.getDetailsQuery(detailObjectType, childObjects) + ' WHERE ';
+
+            // If necessary, filter the Detail records by either ContactId or AccountId
+            if (this.recordIds != null && this.recordIds.size() > 0) {
+                String keyField = SObjectType.OpportunityContactRole.fields.ContactId.getName();
+                if (this.hasContactSoftCreditRollups == false) {
+                    keyField = 'Contact.AccountId';
+                }
+                query += keyField + ' IN :recordIds';
+            }
+
+            // Filter out primary contacts and set the sort order to Account, Contact, Opp
```

Comment:
Reminder that we want to rollup as soft credits OCRs with IsPrimary==true when Opportunity.Account.npe01__SYSTEMISINDIVIDUAL__c = false

Reply:
That will be in the next PR.